<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>634</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>GOF - Game of Life</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="3" column="5">
     <spacer name="horizontalSpacer">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="3" column="0">
     <widget class="QLabel" name="label">
      <property name="text">
       <string>FPS</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="3" column="4">
     <widget class="QGroupBox" name="optionsBox">
      <property name="title">
       <string>Options</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QCheckBox" name="modifyCheckBox">
         <property name="text">
          <string>Modify</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="agingBox">
         <property name="text">
          <string>Aging</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="3" column="1">
     <widget class="QSlider" name="FrameSlider">
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
      <property name="value">
       <number>1</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item row="2" column="13" colspan="2">
     <widget class="QPushButton" name="loadButton">
      <property name="text">
       <string>Load</string>
      </property>
     </widget>
    </item>
    <item row="3" column="8">
     <widget class="QPushButton" name="resetButton">
      <property name="minimumSize">
       <size>
        <width>70</width>
        <height>40</height>
       </size>
      </property>
      <property name="text">
       <string>Reset</string>
      </property>
      <property name="iconSize">
       <size>
        <width>32</width>
        <height>32</height>
       </size>
      </property>
      <property name="shortcut">
       <string>R</string>
      </property>
     </widget>
    </item>
    <item row="3" column="10">
     <widget class="QPushButton" name="playPauseButton">
      <property name="minimumSize">
       <size>
        <width>40</width>
        <height>40</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>40</width>
        <height>40</height>
       </size>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="iconSize">
       <size>
        <width>32</width>
        <height>32</height>
       </size>
      </property>
      <property name="shortcut">
       <string>Space</string>
      </property>
     </widget>
    </item>
    <item row="3" column="2">
     <widget class="QLCDNumber" name="FrameLCD">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>30</height>
       </size>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="lineWidth">
       <number>2</number>
      </property>
      <property name="digitCount">
       <number>3</number>
      </property>
      <property name="value" stdset="0">
       <double>1.000000000000000</double>
      </property>
     </widget>
    </item>
    <item row="3" column="6">
     <spacer name="horizontalSpacer_3">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="0" column="0" rowspan="3" colspan="13">
     <widget class="WorldViewer" name="worldViewer">
      <property name="minimumSize">
       <size>
        <width>640</width>
        <height>360</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
     </widget>
    </item>
    <item row="3" column="7">
     <spacer name="horizontalSpacer_2">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="1" column="13" colspan="2">
     <widget class="QPushButton" name="saveButton">
      <property name="text">
       <string>Save</string>
      </property>
     </widget>
    </item>
    <item row="0" column="13" colspan="2">
     <widget class="QGroupBox" name="statisticsBox">
      <property name="minimumSize">
       <size>
        <width>200</width>
        <height>0</height>
       </size>
      </property>
      <property name="title">
       <string>Statistics</string>
      </property>
      <widget class="QLabel" name="generationLabel">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>71</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Generation:</string>
       </property>
      </widget>
      <widget class="QLabel" name="currentPopulationLabel">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>111</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Current Population:</string>
       </property>
      </widget>
      <widget class="QLabel" name="overallPopulationlabel">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>111</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Overall Deaths:</string>
       </property>
      </widget>
      <widget class="QLabel" name="generationNumberLabel">
       <property name="geometry">
        <rect>
         <x>110</x>
         <y>20</y>
         <width>91</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>0</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="currentPopulationNumber">
       <property name="geometry">
        <rect>
         <x>110</x>
         <y>40</y>
         <width>91</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>0</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="overallPopulationNumber">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>60</y>
         <width>71</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>0</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="updateTimeLabel">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>80</y>
         <width>111</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Update Time (ms):</string>
       </property>
      </widget>
      <widget class="QLabel" name="updateTimeNumber">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>80</y>
         <width>71</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>0</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1280</width>
     <height>20</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">background: rgb(220,220,220)</string>
   </property>
   <widget class="QMenu" name="menuGame_Of_Life">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menuNew_2">
     <property name="title">
      <string>New...</string>
     </property>
     <addaction name="actionEmpty_World"/>
     <addaction name="actionRandom_World"/>
    </widget>
    <addaction name="menuNew_2"/>
    <addaction name="actionLoad_World"/>
    <addaction name="separator"/>
    <addaction name="actionSave_World"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionZoom_In"/>
    <addaction name="actionZom_Out"/>
    <addaction name="separator"/>
    <addaction name="actionFull_View"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Options</string>
    </property>
    <widget class="QMenu" name="menuModify">
     <property name="title">
      <string>Modify</string>
     </property>
     <addaction name="actionEnableModify"/>
     <addaction name="actionDisableModify"/>
    </widget>
    <widget class="QMenu" name="menuAging">
     <property name="title">
      <string>Aging</string>
     </property>
     <addaction name="separator"/>
     <addaction name="actionEnableAging"/>
     <addaction name="actionDisableAging"/>
    </widget>
    <addaction name="menuAging"/>
    <addaction name="menuModify"/>
   </widget>
   <addaction name="menuGame_Of_Life"/>
   <addaction name="menuView"/>
   <addaction name="menuTools"/>
  </widget>
  <action name="actionZoom_In">
   <property name="text">
    <string>Zoom In</string>
   </property>
   <property name="shortcut">
    <string>+</string>
   </property>
  </action>
  <action name="actionZom_Out">
   <property name="text">
    <string>Zom Out</string>
   </property>
   <property name="shortcut">
    <string>-</string>
   </property>
  </action>
  <action name="actionLoad_World">
   <property name="text">
    <string>Load World</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave_World">
   <property name="text">
    <string>Save World</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionFull_View">
   <property name="text">
    <string>Full View</string>
   </property>
  </action>
  <action name="actionEnableModify">
   <property name="text">
    <string>Enable</string>
   </property>
  </action>
  <action name="actionDisableModify">
   <property name="text">
    <string>Disable</string>
   </property>
  </action>
  <action name="actionEnableAging">
   <property name="text">
    <string>Enable</string>
   </property>
  </action>
  <action name="actionDisableAging">
   <property name="text">
    <string>Disable</string>
   </property>
  </action>
  <action name="actionEmpty_World">
   <property name="text">
    <string>Empty World</string>
   </property>
  </action>
  <action name="actionRandom_World">
   <property name="text">
    <string>Random World</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>WorldViewer</class>
   <extends>QGraphicsView</extends>
   <header>worldviewer.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
